### PR TITLE
First functional orb.yaml

### PR DESCRIPTION
**Pull request by stoiev**

### CODE REVIEW

File: src/orb.yaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  localhost-parameter: &localhost-parameter
+    description: host
+    type: string
+    default: localhost
+  wait-for-attributes: &wait-for-attributes
+    seconds_between_retries: <<parameters.seconds_between_retries>>
+    timeout: <<parameters.timeout>>
+    description: <<parameters.description>>
+
+description: Waits for things
```
Comment (Evnsan) [inline src/orb.yaml]: Maybe we can change it to something like:
*Wait until the execution of the command argument returns success(0) or until timeout*

File: src/orb.yaml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    steps:
+      - sh-command:
+          <<: *wait-for-attributes
+          sh_command: >-
+            printf "PING\r\n" |
+            nc -q 1 <<parameters.host>> <<parameters.port>> |
+            grep PONG
+
+  cassandra:
+    description: Waits for redis to be ready
```
Comment (Evnsan) [inline src/orb.yaml]: Waits for **cassandra** to be ready

File: src/orb.yaml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        default: 4572
+    steps:
+      - sh-command:
+          <<: *wait-for-attributes
+          sh_command: >-
+            curl --silent --fail
+            "http://<<parameters.host>>:<<parameters.port>>"
+
+  redis:
+    description: Waits for redis to be ready
```
Comment (Evnsan) [inline src/orb.yaml]: Why we change the description from **being up** to **be ready**?

File: src/orb.yaml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        description: redis description
+        type: string
+        default: Redis
+      port:
+        <<: *port-parameter
+        default: 6379
+    steps:
+      - sh-command:
+          <<: *wait-for-attributes
+          sh_command: >-
```
Comment (Evnsan) [inline src/orb.yaml:143]: :+1:

File: src/orb.yaml [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+          command: |
+            timeout <<parameters.timeout>> sh -c \
+              'until <<parameters.sh_command>>; do
+                echo "Waiting for <<parameters.description>> ..."
+                sleep <<parameters.seconds_between_retries>>
+              done'
+
+  port:
+    description: Waits for a port being up
+    parameters: &wait-for-port-parameters
```
Comment (Evnsan) [inline src/orb.yaml]: This is not a request but a suggestion. I usually like to declare achors in a more global key, variables maybe.
So it is easier to find the declaration whitout using find command. Just saying

File: src/orb.yaml [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+            printf "PING\r\n" |
+            nc -q 1 <<parameters.host>> <<parameters.port>> |
+            grep PONG
+
+  cassandra:
+    description: Waits for redis to be ready
+    parameters:
+      <<: *wait-for-port-parameters
+      description:
+        description: redis description
```
Comment (Evnsan) [inline src/orb.yaml]: **casssandra** description...

File: src/orb.yaml [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+version: 2.1
+
+definitions:
+  port-parameter: &port-parameter
+    description: port
+    type: integer
+  localhost-parameter: &localhost-parameter
+    description: host
+    type: string
+    default: localhost
```
Comment (Evnsan) [inline src/orb.yaml:11]: Are you sure that localhost works? I tried it with application tests and it only works with *cassandra* instead *localhost*.

**Review by Evnsan — CHANGES_REQUESTED** [review]

File: src/orb.yaml (hunk 2 above)
Comment (stoiev) [reply src/orb.yaml]: Thanks!

File: src/orb.yaml (hunk 5 above)
Comment (stoiev) [reply src/orb.yaml]: I have refactored the file and all definitions are inside `definitions` key.

It's definitively better (I toke `definitions` name from OpenAPI's specs section that have the same objective. I think that is more precise that `variables`)

File: src/orb.yaml (hunk 7 above)
Comment (stoiev) [reply src/orb.yaml:11]: Yes. See Wait * commands in this [build](https://circleci.com/gh/Cobliteam/alexstrasza-core/211)

File: src/orb.yaml (hunk 6 above)
Comment (stoiev) [reply src/orb.yaml]: Thanks!

File: src/orb.yaml (hunk 7 above)
Comment (Evnsan) [reply src/orb.yaml:11]: ok

File: src/orb.yaml (hunk 3 above)
Comment (stoiev) [reply src/orb.yaml]: Normalized to  *be ready*

File: src/orb.yaml (hunk 1 above)
Comment (stoiev) [reply src/orb.yaml]: Changed to _Waits for things testable with a sh command_

**Review by Evnsan — APPROVED** [review]

**Pull request merged**